gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala

```diff
@@ -227,12 +227,20 @@ trait GlutenTestsTrait extends GlutenTestsCommonTrait {
     } else {
       logInfo("Has unsupported data type, fall back to vanilla spark.\n")
     }
-    if (!(checkResult(result.head.get(0), expected, expression.dataType, expression.nullable)
-      || checkResult(
-      CatalystTypeConverters.convertToCatalyst(result.head.get(0)),
-      CatalystTypeConverters.convertToCatalyst(expected),
-      expression.dataType,
-      expression.nullable))) {
+    val resultToCompare = result.head.get(0) match {
+      case decimal: java.math.BigDecimal =>
+        // Java.math.BigDecimal causes precision < scale issue.
```

Comment:
This fix seems same with [PR#1697](https://github.com/oap-project/gluten/pull/1697). Can you try locally if PR#1697 can fix this problem? If so, we can merge PR#1697 firstly.

Reply:
Yes, they are for the same issue. Will rebase after PR#1697 being merged. Thanks for pointing this out!

Reply:
[PR#1735](https://github.com/oap-project/gluten/pull/1735) merged. And you can rebase it.

Reply:
Rebased. Could you review again?